bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/factory/SmartherHandlerFactory.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smarther.internal.factory;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.auth.client.oauth2.OAuthFactory;
+import org.eclipse.smarthome.core.scheduler.CronScheduler;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.openhab.binding.smarther.internal.SmartherBindingConstants;
+import org.openhab.binding.smarther.internal.account.SmartherAccountService;
+import org.openhab.binding.smarther.internal.handler.SmartherBridgeHandler;
+import org.openhab.binding.smarther.internal.handler.SmartherDynamicStateDescriptionProvider;
+import org.openhab.binding.smarther.internal.handler.SmartherModuleHandler;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@code SmartherHandlerFactory} class is responsible for creating things and thing handlers.
+ *
+ * @author Fabio Possieri - Initial contribution
+ */
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.smarther")
+@NonNullByDefault
+public class SmartherHandlerFactory extends BaseThingHandlerFactory {
+
+    private @NonNullByDefault({}) OAuthFactory oAuthFactory;
+    private @NonNullByDefault({}) SmartherAccountService authService;
+    private @NonNullByDefault({}) HttpClient httpClient;
+    private @NonNullByDefault({}) CronScheduler cronScheduler;
+    private @NonNullByDefault({}) SmartherDynamicStateDescriptionProvider dynamicStateDescriptionProvider;
```

Comment:
Use of `@NonNullByDefault({})` should be avoided wherever possible. I suggest populating these fields in through the SmartherHandlerFactory constructor and then making these fields final.